### PR TITLE
fix: reference to the moodle_org_token secret

### LINKED ISSUE #40: Secrets input field not handled correctly
The workflow is not valid. In .github/workflows/ci.yml (Line: 8, Col: 11): Error from called workflow catalyst/catalyst-moodle-workflows/.github/workflows/ci.yml@main (Line: 35, Col: 9): Unexpected value 'default' catalyst/catalyst-moodle-workflows/.github/workflows/ci.yml@main (Line: 181, Col: 9): Unrecognized named-value: 'secrets'. Located at position 107 within expression: needs.pre_job.outputs.should_skip != 'true' && needs.prepare_matrix.outputs.release_required == 'true' && secrets.MOODLE_ORG_TOKEN != ''

### PR DESCRIPTION
**Pull request by keevan**
- Secrets cannot be used direcly in a conditional
- Workaround is to pass it into a step in the ENV, and output some flag
  which can be used as a check thereafter
- Also removed the default value for the secrets input, since it was an
  invalid field/property

Resolves #40

### CODE REVIEW

**Comment by keevan** [comment]
Tested here: https://github.com/catalyst/moodle-tool_dataflows/runs/6930584452?check_suite_focus=true

Using a draft PR pinned at the workflow branch: https://github.com/catalyst/moodle-tool_dataflows/pull/211

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -178,11 +177,21 @@ jobs:
...
+        if: ${{ env.SECRET_TO_CHECK != '' }}
+        run: echo "::set-output name=has-secrets::true"
+
       - name: Run plugin release
+        if: steps.check-secrets.outputs.has-secrets != ''
         uses: catalyst/catalyst-moodle-workflows/.github/plugin/release@main
         with:
           plugin_name: ${{ needs.prepare_matrix.outputs.component }}
-          moodle_org_token: ${{ secrets.MOODLE_ORG_TOKEN }}
+          moodle_org_token: ${{ secrets.moodle_org_token }}
```
Comment (cameron1729) [inline .github/workflows/ci.yml:197]: Why the change in case?

**Comment by cameron1729** [comment]
If I understand correctly, this workaround doesn't entail a documentation change, right? That is, someone using this reusable workflow still only has to add the secret to their repo, and nothing else?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (keevan) [reply .github/workflows/ci.yml:197]: It's using the locally scoped secret I believe. This is referenced at the top of the workflow file, and in other examples I've seen. I think this might have been something I've missed initially when I added it.

The uppercase one refers to the secret coming out from the org / plugin settings, so that again, is different.

**Comment by keevan** [comment]
> If I understand correctly, this workaround doesn't entail a documentation change, right? That is, someone using this reusable workflow still only has to add the secret to their repo, and nothing else?

Yes no changes to usage, just fixing an issue with the check / syntax which caused the whole workflow to stop running - an example here: https://github.com/catalyst/moodle-tool_dataflows/actions/runs/2513227928

**Comment by cameron1729** [comment]
LGTM, and the test convinces me it's now working.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (keevan) [reply .github/workflows/ci.yml:197]: Further backed up by the usage from the plugin repo + readme docs:

image

Source: https://github.com/catalyst/moodle-tool_crawler/pull/161/files
